packages/react-dom-bindings/src/client/ReactInputSelection.js

```diff
@@ -56,9 +56,12 @@ function isSameOriginFrame(iframe) {
   }
 }
 
-function getActiveElementDeep() {
+function getActiveElementDeep(containerInfo) {
   let win = window;
-  let element = getActiveElement();
+  if (containerInfo?.ownerDocument?.defaultView) {
+    win = containerInfo.ownerDocument.defaultView
+  }
```

Comment:
Note that we try to minimize optional chaining in the runtime because of how it expands when transpiled, and this case does it twice. Maybe:

```js
let win = containerInfo?.ownerDocument?.defaultView ?? window;
```

Reply:
Done in https://github.com/facebook/react/pull/30951/commits/edd9d8a33cdec0c01e3d4dbdcd0b1dc592d0aa83